atlasdb-config/src/main/java/com/palantir/atlasdb/factory/TransactionManagers.java

```diff
@@ -337,13 +361,21 @@ private TransactionManager serializableInternal(@Output List<AutoCloseable> clos
         MetricsManager metricsManager = MetricsManagers.of(globalMetricsRegistry(), globalTaggedMetricRegistry());
 
         AtlasDbRuntimeConfig defaultRuntime = AtlasDbRuntimeConfig.defaultRuntimeConfig();
-        Supplier<AtlasDbRuntimeConfig> runtimeConfigSupplier =
-                () -> runtimeConfigSupplier().get().orElse(defaultRuntime);
+        Refreshable<AtlasDbRuntimeConfig> runtime = runtimeConfig()
+                .orElseGet(() -> {
+                    ScheduledExecutorService refreshableExecutor = PTExecutors.newSingleThreadScheduledExecutor();
```

Comment:
From my reading of the code, this will not get closed on TransactionManager#close. As the code is written, the closeables list is only cleaned up on exception.

Reply:
See https://github.com/palantir/atlasdb/pull/4723